src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -528,6 +528,17 @@ CreateDistributedTableConcurrently(Oid relationId, char *distributionColumnName,
 		colocatedTableId = ColocatedTableId(colocationId);
 	}
 
+	List *workerNodeList = DistributedTablePlacementNodeList(NoLock);
+	int32 workerNodeCount = list_length(workerNodeList);
+	if (ShardReplicationFactor > workerNodeCount)
+	{
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						errmsg("replication_factor (%d) exceeds number of worker nodes "
+							   "(%d)", ShardReplicationFactor, workerNodeCount),
+						errhint("Add more worker nodes or try again with a lower "
```

Comment:
a bit misleading since we only support replication factor 1 in this case, maybe something like:

errmsg("no worker nodes are available for placing shards")